earn/src/components/borrow/modal/tab/AddUniswapNFTAsCollateralTab.tsx

```diff
@@ -0,0 +1,381 @@
+    );
+  }, [
+    uniswapNFTPosition.tickLower,
+    uniswapNFTPosition.tickUpper,
+    uniswapNFTPosition.liquidity,
+    marginAccount.sqrtPriceX96,
+    uniswapNFTPosition.token1.decimals,
+  ]);
+
+  return (
+    <UniswapNFTPositionButtonWrapper onClick={onClick} active={isActive}>
+      <div className='flex items-center gap-4'>
+        <TokenPairIcons
+          token0IconPath={token0.iconPath}
+          token1IconPath={token1.iconPath}
+          token0AltText={`${token0.name}'s Icon`}
+          token1AltText={`${token1.name}'s Icon`}
+        />
+        <Display size='S' weight='semibold'>
+          {token0.ticker} / {token1.ticker}
+        </Display>
+      </div>
+      <div className='flex flex-col items-start gap-1'>
+        <Text size='M' weight='medium' color={SECONDARY_COLOR}>
+          Total Liquidity:
+        </Text>
+        <Display size='S' weight='semibold'>
+          {truncateDecimals(liquidityAmount.toString(), 6)} {token1.ticker}
+        </Display>
+      </div>
+      <div className='flex items-center gap-4'>
+        <Text size='S' color={SECONDARY_COLOR}>
+          Min: {truncateDecimals(minPrice.toString(), 3)} {token1.ticker} per {token0.ticker} - Max:{' '}
+          {truncateDecimals(maxPrice.toString(), 3)} {token1.ticker} per {token0.ticker}
+        </Text>
+      </div>
+    </UniswapNFTPositionButtonWrapper>
+  );
+}
+
+type AddUniswapNFTAsCollateralButtonProps = {
+  marginAccount: MarginAccount;
+  uniswapNFTPosition: [number, UniswapNFTPosition];
+  userAddress: string;
+  setIsOpen: (open: boolean) => void;
+  setPendingTxn: (result: SendTransactionResult | null) => void;
+};
+
+function AddUniswapNFTAsCollateralButton(props: AddUniswapNFTAsCollateralButtonProps) {
+  const { marginAccount, uniswapNFTPosition, setIsOpen, setPendingTxn } = props;
+  const { activeChain } = useContext(ChainContext);
+
+  const [isPending, setIsPending] = useState(false);
+  const [approvingTxn, setApprovingTxn] = useState<SendTransactionResult | null>(null);
+
+  // MARK: Read/write hooks for Router's allowance --------------------------------------------------------------------
+  const { refetch: refetchGetApprovedData, data: getApprovedData } = useContractRead({
+    address: UNISWAP_NONFUNGIBLE_POSITION_MANAGER_ADDRESS,
+    abi: erc721ABI,
+    functionName: 'getApproved',
+    args: [BigNumber.from(uniswapNFTPosition[0].toFixed(0))] as const,
+    chainId: activeChain.id,
+  });
+  const { writeAsync: writeApproveAsync } = useContractWrite({
+    address: UNISWAP_NONFUNGIBLE_POSITION_MANAGER_ADDRESS,
+    abi: erc721ABI,
+    functionName: 'approve',
+    mode: 'recklesslyUnprepared',
+    chainId: activeChain.id,
+  });
+
+  const data = ethers.utils.defaultAbiCoder.encode(
+    ['uint256', 'int24', 'int24', 'int128', 'uint144'],
+    [
+      uniswapNFTPosition[0],
+      uniswapNFTPosition[1].tickLower,
+      uniswapNFTPosition[1].tickUpper,
+      `-${uniswapNFTPosition[1].liquidity.toString()}`,
```

Comment:
I believe JSBI's `toString()` excepts a radix arg. If so, we should specify base 10.